awswrangler/timestream.py

```diff
@@ -240,21 +249,22 @@ def write(
     _logger.debug("cols_names: %s", cols_names)
     batches: List[List[Any]] = _utils.chunkify(lst=_df2list(df=df[cols_names]), max_length=100)
     _logger.debug("len(batches): %s", len(batches))
-    with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
-        res: List[List[Any]] = list(
-            executor.map(
-                _write_batch,
-                itertools.repeat(database),
-                itertools.repeat(table),
-                itertools.repeat(cols_names),
-                itertools.repeat(measure_cols_names),
-                itertools.repeat(measure_types),
-                itertools.repeat(version),
-                batches,
-                itertools.repeat(_utils.boto3_to_primitives(boto3_session=boto3_session)),
-            )
-        )
-        return [item for sublist in res for item in sublist]
+
+    executor = _get_executor(use_threads=use_threads)
+
+    res = executor.map(
+        _write_batch,
+        boto3_session,
+        itertools.repeat(database),
+        itertools.repeat(table),
+        itertools.repeat(cols_names),
+        itertools.repeat(measure_cols_names),
+        itertools.repeat(measure_types),
+        itertools.repeat(version),
+        batches,
+    )
+
+    return [item for sublist in res for item in sublist]
 
 
```

Comment:
Could we add a timestream [load test](https://github.com/awslabs/aws-data-wrangler/tree/release-3.0.0/load_tests) to ensure that this code actually scales?